### PR TITLE
refactor: use express-session instead of cookie-session and cookie-parser

### PR DESCRIPTION
**Pull request by neilsong**
`express-session` is a superset of the functionality of `cookie-session`. It removes the dependency on `cookie-parser` and is required to solve the login issue in #1546

### CODE REVIEW

File: api/src/server.ts [hunk 1]
```diff
@@ -62,19 +61,25 @@ app.use((req, _, next) => {
 
 // Setup sessions.
 app.use(
-  cookieSession({
+  session({
```
Comment (neilsong) [inline api/src/server.ts:64]: Need some help on solving the eslint error here :)

**Comment by Josh-Cena** [comment]
@neilsong FYI: whenever you add a new module, make sure you have its `@types` (unless the package provides native types). If you remove a module, remove its `@types` too. The `unsafe-call` is because that `session` function has no types so it becomes `any`.

**Comment by neilsong** [comment]
Gotcha, thank you so much!

**Comment by Josh-Cena** [comment]
Could you tell me a bit more about what doesn't work with `cookie-parser`?

**Comment by neilsong** [comment]
express-session just takes care of cookie-parser, I assume they just implemented all the functionality.

**Comment by Josh-Cena** [comment]
Have you tested it? If so I'm not going to duplicate the effort.

**Comment by neilsong** [comment]
I've tested it - basic auth functionality, loading courses, and adding them works. Outside of that, I'm not sure.

**Review by Josh-Cena — APPROVED** [review]

**Pull request merged**